core/ssz.go

```diff
@@ -42,7 +43,7 @@ func (b VersionedSignedBeaconBlock) MarshalSSZ() ([]byte, error) {
 
 // MarshalSSZTo ssz marshals the VersionedSignedBeaconBlock object to a target array.
 func (b VersionedSignedBeaconBlock) MarshalSSZTo(buf []byte) ([]byte, error) {
-	return marshalSSZVersionedTo(buf, b.Version, b.sszValFromVersion)
+	return marshalSSZVersionedTo(buf, eth2util.DataVersion(b.Version), b.sszValFromVersion)
```

Comment:
note that casting isn't enough, we need to ensure the numbers we are working with are aligned with pre-v0.18.0

Reply:
this has been refactored to convert instead of cast.